avt_vimba_camera/src/avt_vimba_camera.cpp

```diff
@@ -1010,4 +1011,45 @@ void AvtVimbaCamera::getCurrentState(diagnostic_updater::DiagnosticStatusWrapper
       break;
   }
 }
+
+bool AvtVimbaCamera::loadCameraSettings(const std::string& filename)
+{
+  stopImaging();
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+  vimba_camera_ptr_->LoadSaveSettingsSetup(VmbFeaturePersistNoLUT, 5, 4);
+  auto err = vimba_camera_ptr_->LoadCameraSettings(filename);
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  startImaging();
+
+  if (err != VmbErrorSuccess)
+  {
+    RCLCPP_ERROR(nh_->get_logger(), "Failed to load camera settings from %s", filename.c_str());
+    return false;
+  }
+  RCLCPP_INFO(nh_->get_logger(), "Loaded camera settings from %s", filename.c_str());
+  return true;
+}
+
+bool AvtVimbaCamera::saveCameraSettings(const std::string& filename)
+{
+  stopImaging();
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+
+  vimba_camera_ptr_->LoadSaveSettingsSetup(VmbFeaturePersistNoLUT, 5, 4);
+  auto err = vimba_camera_ptr_->SaveCameraSettings(filename);
+
+  std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  startImaging();
+
+  if (err != VmbErrorSuccess)
+  {
+    RCLCPP_ERROR(nh_->get_logger(), "Failed to save camera settings to %s", filename.c_str());
```

Comment:
I've had this fail because the filename I specified was "settings.txt" instead of "settings.xml". 
Maybe add a check at the start of the function to make sure the string ends with ".xml", and output a different error/failure that tells the user to use an xml file?

Reply:
Generating XML configuration files from Vimba Viewer is really easy and is not prone to mistakes (sliders etc.). Also, in one of my project we are using Basler cameras and loading configs such files simplified our architecture. So I did this the same for AVT cameras ;)

Reply:
@icolwell-as one question. What code style do you use in this project? [ROS2](https://docs.ros.org/en/foxy/Contributing/Code-Style-Language-Versions.html#id1) advice to use Google C++ code style with 100 line width.

Reply:
solved this in mono_camera_node